dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -405,4 +412,41 @@ private boolean isClassAccessibleFrom(JavaClass javaClass, String sourceClassNam
       return false;
     }
   }
+
+  /**
+   * Returns an {@code Optional} describing the symbol problem in the parent classes or interfaces
+   * for {@code baseClassName}, if any of them are missing; otherwise an empty {@code Optional}.
+   */
+  private Optional<SymbolProblem> findParentSymbolProblem(String baseClassName) {
+    Queue<String> queue = new ArrayDeque<>();
+    queue.add(baseClassName);
+    while (!queue.isEmpty()) {
+      String className = queue.remove();
+      if (Object.class.getName().equals(className)) {
+        continue; // java.lang.Object is the root of the inheritance tree
+      }
+      JavaClass baseClass;
```

Comment:
This could be pushed into the try block

Reply:
No, I tried that but this declaration cannot be in the try block. because baseClass is used after try block at Line 439.